apps/chat/models.py

```diff
@@ -95,6 +100,22 @@ class ChatMessage(BaseModel, TaggedModelMixin, UserCommentsMixin):
     class Meta:
         ordering = ["created_at"]
 
+    @classmethod
+    def make_summary_message(cls, content):
+        """A 'summary message' is a special message only ever exists in memory. It is
+        not saved to the database. It is used to represent the summary of a chat up to a certain point."""
+        return ChatMessage(message_type=ChatMessageType.SYSTEM, content=content, metadata={"summary": True})
+
+    def save(self, *args, **kwargs):
+        if self.is_summary:
+            raise ValueError("Cannot save a summary message")
+        super().save(*args, **kwargs)
+
+    def get_summary_message(self):
+        if not self.summary:
+            raise ValueError("Message does not have a summary")
```

Comment:
Out of curiosity, what is the reason for raising a `ValueError` rather than just returning an empty string?

Reply:
Because I think it's better to make sure this isn't called when it shouldn't be than to to check for `None` when calling this or else expecting an `AttributeError`